pymatgen/io/cif.py

```diff
@@ -1099,7 +1103,14 @@ def get_matching_coord(coord):
                 all_labels = None  # type: ignore
 
             struct = Structure(lattice, all_species, all_coords, site_properties=site_properties, labels=all_labels)
-
+            if skip_occu_checks:
+                struct_2 = Structure(
+                    lattice, all_species, all_coords, site_properties=site_properties, labels=all_labels
+                )
+                for idx in range(len(struct_2)):
+                    struct_2[idx] = PeriodicSite(
+                        all_species_noedit[idx], all_coords[idx], lattice, properties=site_properties, skip_checks=True
+                    )
```

Comment:
This looks needlessly complicated. What's the reason for creating `struct_2`?

Reply:
I think the reason struct_2 was there is no longer in cif.py. I've updated the file so that it doesn't make a new object.